db/database.go

```diff
@@ -1217,6 +1252,77 @@ func (db *Database) UpdateAllDocChannels() (int, error) {
 		}
 		startSeq = highSeq + 1
 	}
+
+	for _, sequence := range unusedSequences {
+		err := db.sequences.releaseSequence(sequence)
+		if err != nil {
+			base.Warnf("Error attempting to release sequence %d. Error %v", sequence, err)
+		}
+	}
+
+	if regenerateSequences {
+		users, roles, err := db.AllPrincipalIDs()
+		if err != nil {
+			return docsChanged, err
+		}
+
+		authr := db.Authenticator()
+		regeneratePrincipalSequences := func(princ auth.Principal, isUser bool) error {
+			nextSeq, err := db.DatabaseContext.sequences.nextSequence()
+			if err != nil {
+				return err
+			}
+			princ.SetSequence(nextSeq)
+
```

Comment:
As discussed, we shouldn't need to update the sequence values in explicitChannels/explicitRoles.  Given that, we could do a simpler version of this that used lower-level bucket ops to just update the sequence value.